### PR TITLE
Update Plotly for Non-Textured Pointclouds

### PR DESCRIPTION
**Pull request by NikV-JS**
Fixed numpy_colors initialization error when include_colors is False.

### Description
A Small Bug fix in the plotly function of pointcloud structure which prevents one from plotting non-textured pointclouds. Error was present in the if-else statement leading to an initialization error.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/gradslam/gradslam/blob/main/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before?
- [x] **Pass all tests**: did you test in local ? `pytest tests/`
- [ ] Unittests: did you add tests for your new functionality?
- [x] Documentations: did you build documentation ? `cd docs/ && sphinx-build . _build`
- [x] Implementation: is your code well commented and follow conventions ? `black gradslam/ && flake8 gradslam/`
- [ ] Docstrings & Typing: has your code documentation and typing ?
- [ ] Update notebooks & documentation if necessary

### gradslam Team
<details>
  <summary>gradslam team workflow</summary>
  
  - [ ] Triage
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `gradslam` design conventions?
  - [ ] Is the documentation complete enough?
  - [ ] Are the tests covering simple and corner cases?
 
</details>

### CODE REVIEW

File: gradslam/structures/pointclouds.py [hunk 1]
```diff
@@ -1340,17 +1340,28 @@ def plotly(
                 torch_colors = torch_colors * 255
             torch_colors = torch.clamp(torch_colors, min=0.0, max=255.0)
             numpy_colors = torch_colors.detach().cpu().numpy().astype("uint8")
-
-        scatter3d = go.Scatter3d(
-            x=numpy_points[..., 0],
-            y=numpy_points[..., 1],
-            z=numpy_points[..., 2],
-            mode="markers",
-            marker=dict(
```
Comment (saryazdi) [inline gradslam/structures/pointclouds.py:1349]: @NikV-JS I'd suggest refactoring it this way: Initializing a dictionary `marker = {"size": point_size}` on line 1328 before the if statement, and adding `marker["color"] =  numpy_colors` on line 1343 within `if self.has_colors and include_colors:`. And here just changing line 1349 to `marker=marker`.

Comment (NikV-JS) [reply gradslam/structures/pointclouds.py:1349]: @saryazdi The suggested changes have been made. marker_dict is initialized, and depending on the include_colors condition, numpy_colors are added to the dictionary.

**Comment by krrish94** [comment]
@NikV-JS Could you please squash your commits so we have a clean history?

File: gradslam/structures/pointclouds.py (hunk 1 above)
Comment (saryazdi) [reply gradslam/structures/pointclouds.py:1349]: @NikV-JS I'd suggest merging line 1334 and 1335 into a single line, i.e. `marker_dict = {"size": point_size}`

**Comment by saryazdi** [comment]
@NikV-JS Can you also lint your code please? From [CONTRIBUTING.md](https://github.com/gradslam/gradslam/blob/main/CONTRIBUTING.md): Make sure your code lints by running `black gradslam/ && flake8 gradslam/` from the project root.

**Comment by krrish94** [comment]
@NikV-JS Please don't run `black` on the entire repo -> only run it on the file that `flake8` reports an error on

**Comment by NikV-JS** [comment]
@krrish94 @saryazdi The code was linted. I ran flake8 gradslam/ from the root, and no errors were reported.

File: gradslam/structures/pointclouds.py (hunk 1 above)
Comment (NikV-JS) [reply gradslam/structures/pointclouds.py:1349]: They have been merged and linted.

**Comment by saryazdi** [comment]
LGTM
